### PR TITLE
fix: CTS bugs in Sysman Device module

### PR DESCRIPTION
**Pull request by vishnu-khanth**
Related-To: VLCLJ-2272

### CODE REVIEW

File: conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp [hunk 1]
```diff
@@ -622,7 +625,15 @@ TEST_F(
     EXPECT_NE(core_device, nullptr);
 #endif
 
-    auto processes = lzt::get_processes_state(device, count);
+    ze_result_t result{};
+    auto processes = lzt::get_processes_state(device, count, result);
+    if (result == ZE_RESULT_ERROR_UNSUPPORTED_FEATURE) {
+      LOG_INFO << "zesDeviceProcessesGetState is Unsupported!";
```
Comment (aviralni) [inline conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp:631]: I think we can put log as  ... Unsupported on this device/driver  for clear msg.

File: utils/test_harness/sysman/src/test_harness_sysman_device.cpp [hunk 2]
```diff
@@ -68,6 +68,12 @@ uint32_t get_processes_count(zes_device_handle_t device) {
   return count;
 }
 
+uint32_t get_processes_count(zes_device_handle_t device, ze_result_t &result) {
+  uint32_t count = 0;
+  result = zesDeviceProcessesGetState(device, &count, nullptr);
```
Comment (pratikbariintel) [inline utils/test_harness/sysman/src/test_harness_sysman_device.cpp:73]: Minor clarification: W.r.t the earlier offline discussion we had, does this zesDeviceProcessesGetState invocation also return Unsupported feature for the integrated device or is it only with line no. 108 ?

File: conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp [hunk 3]
```diff
@@ -680,15 +691,35 @@ TEST_F(
...
         engine_type = process.engines;
       }
       processes.clear();
+      elapsed_seconds = std::chrono::system_clock::now() - start;
     }
 
     thread.join();
+    if (result == ZE_RESULT_ERROR_UNSUPPORTED_FEATURE) {
+      // proceeed to next device
+      continue;
```
Comment (pratikbariintel) [inline conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp:717]: @AshwinKumarKulkarni should we have this condition here to proceed for other devices ?

**Review by harinik124 — COMMENTED** [review]
Looks fine.

File: utils/test_harness/sysman/src/test_harness_sysman_device.cpp (hunk 2 above)
Comment (vishnu-khanth) [reply utils/test_harness/sysman/src/test_harness_sysman_device.cpp:73]: zesDeviceProcessesGetState(device, &count, nullptr) also return unsuppoted for iGPU's.
In the newly added overloaded function if the first invocation of zesDeviceProcessesGetState fails, then i am simply returning at line 98

File: conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp (hunk 1 above)
Comment (vishnu-khanth) [reply conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp]: Updated at line 697

**Review by pratikbariintel — DISMISSED** [review]
lgtm !

**Review by AshwinKumarKulkarni — DISMISSED** [review]

File: conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp [hunk 4]
```diff
@@ -680,16 +684,35 @@ TEST_F(
...
         engine_type = process.engines;
       }
       processes.clear();
+      elapsed_seconds = std::chrono::system_clock::now() - start;
     }
 
     thread.join();
-    EXPECT_EQ(engine_type, ZES_ENGINE_TYPE_FLAG_COMPUTE);
+    if (elapsed_seconds >= std::chrono::seconds{30}) {
+      FAIL() << "TimeOut of 30 seconds elapsed while waiting for engine type "
```
Comment (harinik124) [inline conformance_tests/sysman/test_sysman_device/src/test_sysman_device.cpp:709]: The Fail message is not clear to me. But if its correct please ignore my comment

**Review by harinik124 — APPROVED** [review]

**Comment by vishnu-khanth** [comment]
FYI, the workaround has been removed.
Now the test is passing consistently in platforms where KMD updates the process state correctly. 
Failing in platforms where KMD doesn't update process state correctly.

**Review by joshuaranjan — APPROVED** [review]

**Review by AshwinKumarKulkarni — APPROVED** [review]

**Pull request merged**